gui/elements/_viewer.py

```diff
@@ -1,87 +1,65 @@
-from PyQt5.QtCore import Qt
-from PyQt5.QtWidgets import QLabel
+from PyQt5.QtCore import Qt, pyqtSignal, QObject
 from PyQt5.QtGui import QCursor, QPixmap
 from .qt import QtViewer
 
-from ..util.misc import (compute_max_shape as _compute_max_shape,
-                         guess_metadata)
 from numpy import clip, integer, ndarray, append, insert, delete
 from copy import copy
 
+
 from os.path import dirname, join, realpath
 dir_path = dirname(realpath(__file__))
 path_cursor = join(dir_path,'qt','icons','cursor_disabled.png')
 
-class Viewer:
+class Viewer(QObject):
     """Viewer containing the rendered scene, layers, and controlling elements
-    including dimension sliders.
-
-    Parameters
-    ----------
-    parent : Window
-        Parent window.
+    including dimension sliders, and control bars for color limits.
 
     Attributes
     ----------
-    camera : vispy.scene.Camera
-        Viewer camera.
+    window : Window
+        Parent window.
     layers : LayerList
         List of contained layers.
-    indices : list
-        Slicing indices controlled by the sliders.
-    window : Window
-        Parent window
-    max_dims : int
-        Maximum dimensions of the contained layers.
-    max_shape : tuple of int
-        Maximum shape of the contained layers.
+    dimensions : Dimensions
+        Contains axes, indices, dimensions and sliders.
+    controlBars : ControlBars
+        Contains contorl bar sliders.
+    camera : vispy.scene.Camera
+        Viewer camera.
     """
-    def __init__(self, window):
+
+    statusChanged = pyqtSignal(str)
+    helpChanged = pyqtSignal(str)
+
+    def __init__(self):
+        super().__init__()
         from ._layer_list import LayerList
-        from ._controls import Controls
+        from ._control_bars import ControlBars
+        from ._dimensions import Dimensions
```

Comment:
Dunno about the other two but `LayerList` is here because it would create a circular import when it imports `Layer` from `gui.layers` while `gui.layers` includes the `add_to_viewer` function which imports `Viewer`.

I haven't really checked/thought this through but it might be able to be removed in #90 where `LayerList` has been moved from `gui.elements` to `gui.layers`.

Reply:
Maybe to resolve this we can place `add_to_viewer` in a `util` module?

Reply:
Agreed - we should think about trying to fix this, but I'd say it can wait till a future PR.